src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -130,6 +110,14 @@ public void Dispose()
         }
     }
 
+    protected abstract bool HasSynchronousComputeFunction { get; }
+    protected abstract bool HasAsynchronousComputeFunction { get; }
+
+    protected abstract T InvokeComputeFunction(CancellationToken cancellationToken);
+    protected abstract Task<T> InvokeComputeFunctionAsync(CancellationToken cancellationToken);
+
+    protected abstract void ResetComputeFunctions();
```

Comment:
💡 Reset seems like it might mean restoring them to a state where they can be called again. `Clear` or `Release` might be better.